File: src/main/java/box/BoxTester.java
package box;

import java.util.Scanner;

public class BoxTester {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Please enter object's length: ");
        int length = scanner.nextInt();
        System.out.print("Please enter object's width: ");
        int width = scanner.nextInt();
        System.out.print("Please enter object's height: ");
        int height = scanner.nextInt();

        BoxThree box3 = new BoxThree(length, width, height);
        BoxFive box5 = new BoxFive(length, width, height);
        if (box3.validate()){
            System.out.println("Please use box3 for your package");
        } else if (box5.validate()) {
            System.out.println("Please use box5 for your package");
        } else {
            System.out.println("There is no suitable boxes for your package");
        }

        /*BoxNumber box3 = new BoxThreeO(length, width, height);
        BoxNumber box5 = new BoxFiveO(length, width, height);
        if (box3.validate())
            System.out.println("box3");
        else if (box5.validate())
            System.out.println("box5");
        else
            System.out.println("Thank you!");*/
    }
}

File: src/main/kotlin/box/BoxTester.kt
package box.kotlin

import java.util.Scanner

fun main() {
    print("Please enter a object's length: ")
    val length = readLine()!!.toFloat()
    print("Please enter a object's width: ")
    val width = readLine()!!.toFloat()
    print("Please a enter object's height: ")
    val height = readLine()!!.toFloat()

    var box3 = BoxTester(length, width, height)
    var box5 = BoxFive(length, width, height)
    if (box3.validate()) println("box3")
    else if (box5.validate()) println("box5")
    else println("Thank you!")
}

class BoxFive(length: Float, width: Float, height: Float):BoxTester(length, width, height){
    override fun validate() = if (length < 39.5 && width < 27.5 && height < 23) true else false
}

open class BoxTester(val length:Float, val width:Float, val height:Float){
    open fun validate()= if (length < 23 && width < 14 && height < 13) true else false
}